azurelinuxagent/ga/exthandlers.py

```diff
@@ -1145,6 +1191,11 @@ def __init__(self, ext_handler, protocol, execution_log_max_size=(10 * 1024 * 10
         self.pkg_file = None
         self.logger = None
         self.set_logger(extension=extension, execution_log_max_size=execution_log_max_size)
+        self._signature_validated = self.get_signature_validated()
```

Comment:
Let's try to avoid calling instance methods from init(), since the object may be not be fully initialized. I think this could simply be initialized to False.

Reply:
I see; there is cascading effect that would require making more members static as well, and possibly also would require changing the way this class is instantiated. Though possible, probably not worth at this point.

One needs to be careful when calling instance members from init. In this case, it is not an issue, because the object is fully initialized when you make the call, but as code evolves somebody may introduce a bug. Let's add a blank line and a comment before invoking the method to call attention that the object must be fully initialized at this point.